### PR TITLE
[docs] Refine navigation setup

### PR DESCRIPTION
**Pull request by mattstein**
## The Problem/Issue/Bug:

As demonstrated by the changes in #4293, the combined section and on-page sidebar navigation can get _in the way_ of browsing a section or page—which is its sole purpose:

FAQ page revisions with current navigation

The problem isn’t just limited to suggested FAQ page revisions, though. The same imbalance is demonstrated on the [Extending and Customizing Environments](https://ddev.readthedocs.io/en/stable/users/extend/customization-extendibility/) page:

Extending page with current navigation

## How this PR Solves The Problem:

This PR makes a few changes to the theme configuration to balance things out and make the navigation more useful for browsing:

1. Separates the on-page table of contents from section navigation by disabling [`navigation.integrate`](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/?h=instant#navigation-integration). This makes it easier to focus on the flow of an entire section, or of a page, without confusing the two or overloading one side of the page—and with a negligible impact on line length.
2. Limits on-page table of contents to two levels ([`toc_depth: 2`](https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/?h=toc_depth#+toc.toc_depth)), which further emphasizes (and encourages!) thoughtful organization without overloading the table of contents.
3. Uses frontmatter to hide the table of contents on pages where it takes visual attention without offering any value.
4. Enables instant loading with [`navigation.instant`](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/?h=instant#instant-loading), which seems to work well and speed up browsing—which translates to better user experience.
5. Enables [`navigation.tracking`](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/?h=instant#anchor-tracking) so the address bar updates with current anchors on scroll.
6. Slightly reorganizes the landing page’s sidebar to include a “Settling In” section meant for post-install adventures. (Currently, installation steps wander into configuration and community details, with an awkward vertical space.)

The result is cleaner, more balanced, and easier to navigate despite varying page organization and length:

FAQ page revisions with improved navigation

Extending page with improved navigation

sidebar-before-and-after

## Manual Testing Instructions:

https://ddev.readthedocs.io/en/stable/developers/testing-docs/#preview-changes

## Automated Testing Overview:

n/a

## Related Issue Link(s):

Part of this overall effort:

- #4188

## Release/Deployment notes:

n/a

<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/4294"></a>

### CODE REVIEW

**Comment by mattstein** [comment]
> I like it! I especially like the "Installing" and "Settling in" sectionals. It's possible though that "Settling in" is too jargony for non-English-first-language folks.

Good point! Maybe “Getting Started” would be more appropriate given that it’s the “Start!” page and we’ve just finished installing?

> Make sure you look at this with a smaller screen, like phone. I find navigating the mkdocs-material menus on the phone complicated, but reading straight through is wonderful. I've taken to reading docs like this on phone just reading and using the "next" on the bottom. Anyway, the right-hand menu you've got on on the web browser is not available at all as far as I know on the phone.

This doesn’t change the mobile navigation UX from its current form, where the table of contents is (thoughtfully, I think) isolated to its own submenu:

Screen Shot 2022-10-14 at 09 29 46 AM

**Comment by rfay** [comment]
Let me know when you're ready to pull.

File: mkdocs.yml [hunk 1]
```diff
@@ -109,11 +110,12 @@ nav:
       - users/install/index.md
       - users/install/docker-installation.md
       - users/install/ddev-installation.md
+    - 'Settling In':
```
Comment (mattstein) [inline mkdocs.yml:113]:
```suggestion
    - 'Getting Started':
```

**Comment by mattstein** [comment]
Just updated “Settling In” to “Getting Started”—pull whenever you’d like!

**Pull request merged**